File: 刷题/剑指offer/用两个栈实现队列.py
# python
class Solution:
    def __init__(self):
        self.stackA=[]  // 第一个栈
        self.stackB=[]  //第二个栈
    def push(self,node):
        self.stackA.append(node)
    def pop(self):
        if self.stackB:  # 如果B中有元素
            return self.stackB.pop()
        elif not self.stackA: # 如果A\B中都没有元素
            return None
        else:  # 如果B中没有元素，A中有元素，就把A中的元素存到B中
            while self.stackA:
                self.stackB.append(self.stackA.pop())
            return self.stackB.pop()

File: 刷题/剑指offer/用两个栈实现队列.cpp
class Solution
{
public:
    void push(int node){
            stack1.push(node)
        }
    int pop(){
        if(stack2.empty())
        {
            while(!stack1.empty())
            {
                stack2.push(stack1.top());
                stack1.pop();
            }
        }
        else
        {
            temp=stack2.pop();
            stack2.pop();
            return temp;
        }
    }
private:
    stack<int> stack1;
    stack<int> stack2;
    int temp;
}